snuba/query/parser/strings.py

```diff
@@ -0,0 +1,36 @@
+from snuba.datasets.tags_column_processor import NESTED_COL_EXPR_RE
+from snuba.query.expressions import Column, Expression, Literal, SubscriptableReference
+from snuba.util import QUOTED_LITERAL_RE
+
+
+def parse_string_to_expr(val: str) -> Expression:
+    """
+    Parses a string expression found in a Snuba Query into an AST expression.
+    This is supposed to work for any non structured expression in the Snuba Query.
+    """
+    match = NESTED_COL_EXPR_RE.match(val)
+    if match:
+        col_name = match[1]
+        key_name = match[2]
+        return SubscriptableReference(
+            # TODO: apply aliases on all AST nodes created during parsing and that do not
+            # already have one. This makes the field names from the original query
+            # (that the client expects in the response) available when building the
+            # Clickhouse query without leaking the logical schema.
+            alias=val,
+            subscriptable_column=Column(None, col_name, None),
+            key=Literal(None, key_name),
+        )
+
+    # TODO: This will use the schema of the dataset/entity to decide if the expression is
+    # a column or a literal.
```

Comment:
Should this also use the logical schema to identify if the column can be subscripted, and throw if not?

Reply:
Yes, that's not the only option, but it seems to me the most appropriate. Her ewe can easily have access to the dataset so such validation can be done.